src/sound/backend/oalsound.cpp

```diff
@@ -2152,20 +2153,33 @@ void OpenALSoundRenderer::UpdateListener(SoundListener *listener)
 				alFilterf(EnvFilters[1], AL_LOWPASS_GAINHF, 1.f);
 
 				// Apply the updated filters on the sources
-				for(uint32_t i = 0;i < ReverbSfx.Size();++i)
+
+				while (schan)
 				{
-					alSourcei(ReverbSfx[i], AL_DIRECT_FILTER, EnvFilters[0]);
-					alSource3i(ReverbSfx[i], AL_AUXILIARY_SEND_FILTER, EnvSlot, 0, EnvFilters[1]);
+					ALuint source = GET_PTRID(schan->SysChannel);
+					if (source && !(schan->ChanFlags & CHAN_UI))
+					{
+						alSourcei(source, AL_DIRECT_FILTER, EnvFilters[0]);
+						alSource3i(source, AL_AUXILIARY_SEND_FILTER, EnvSlot, 0, EnvFilters[1]);
+					}
+					schan = schan->NextChan;
 				}
 			}
 
-			for(uint32_t i = 0;i < ReverbSfx.Size();++i)
-				alSourcef(ReverbSfx[i], AL_PITCH, PITCH_MULT);
+			
+			while (schan)
+			{
+				ALuint source = GET_PTRID(schan->SysChannel);
+				if (source && !(schan->ChanFlags & CHAN_UI))
+					alSourcef(source, AL_PITCH, schan->Pitch / 128.0f);
+				schan = schan->NextChan;
+			}
```

Comment:
Since `schans` isn't reassigned for this loop, it remains null from the last loop. The loop lower down also has this issue. This one's also missing the `PITCH_MULT` for the underwater pitch adjustment.

Reply:
Before I replace the part, you gave me different code for that instead of PITCH_MULT. Should I replace that?

Reply:
Done.